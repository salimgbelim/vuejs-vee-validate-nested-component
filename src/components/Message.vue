<template>
  <li class="message" style="margin-top:10px;">{{message}} -- {{ author }}</li>
</template>

<script>
  export default {
    name: 'Message',
    props: {
      message: {
        type: String,
        required: true,
        validator: message => message.length > 1
      },
      author: {
        type: String,
        default: 'Paco'
      }
    }
  }
</script>
