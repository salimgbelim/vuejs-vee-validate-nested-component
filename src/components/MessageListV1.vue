<template>
  <ul>
    <message :message="message" v-for="message in messages">
    </message>
  </ul>
</template>

<script>

  import Message from './Message';

  export default {
    name: 'list',
    props: ['messages'],
    components: {
      Message
    }
  }
</script>
