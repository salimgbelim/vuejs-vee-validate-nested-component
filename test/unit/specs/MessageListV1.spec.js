import {mount} from 'vue-test-utils';
import MessageListV1 from '@/components/MessageListV1';
import Message from '@/components/Message';

describe('MessageListV1.test.js', () => {

  let cmp;

  beforeEach(() => {

    cmp = mount(MessageListV1, {
      propsData: {
        messages: ['Cat']
      }
    });

  });

  it('has received ["Cat"] as the message property', () => {
    expect(cmp.vm.messages).toEqual(['Cat']);
  });

  it('is a MessageList component', () => {
    expect(cmp.is(MessageListV1)).toBe(true);

    // or with css selector
    expect(cmp.is('ul')).toBe(true);
  });

  it('contains a Message component', () => {
    expect(cmp.contains(Message)).toBe(true);

    // or with css selector
    expect(cmp.contains('.message')).toBe(true);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
