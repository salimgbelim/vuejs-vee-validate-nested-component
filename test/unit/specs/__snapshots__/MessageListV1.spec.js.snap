// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageListV1.test.js has the expected html structure 1`] = `
<ul>
  <li
    class="message"
  >
    Cat
  </li>
</ul>
`;
